lib/Sema/MiscDiagnostics.cpp

```diff
@@ -1434,11 +1434,53 @@ void swift::fixItAvailableAttrRename(TypeChecker &TC,
   });
 
   if (auto args = dyn_cast<TupleShuffleExpr>(argExpr)) {
-    if (!args->getVariadicArgs().empty()) {
-      // FIXME: Support variadic arguments.
+    argExpr = args->getSubExpr();
+
+    // Coerce the `argumentLabelIDs` to the user supplied arguments.
+    // e.g:
+    //   @available(.., renamed: "new(w:x:y:z:)")
+    //   func old(a: Int, b: Int..., c: String="", d: Int=0){}
+    //   old(a: 1, b: 2, 3, 4, d: 5)
+    // coerce
+    //   argumentLabelIDs = {"w", "x", "y", "z"}
+    // to
+    //   argumentLabelIDs = {"w", "x", "", "", "z"}
+    auto elementMap = args->getElementMapping();
+    if (elementMap.size() != argumentLabelIDs.size()) {
+      // Mismatched lengths; give up.
       return;
     }
-    argExpr = args->getSubExpr();
+    auto I = argumentLabelIDs.begin();
+    for (auto suffleIdx : elementMap) {
+      switch (suffleIdx) {
+      case TupleShuffleExpr::DefaultInitialize:
+      case TupleShuffleExpr::CallerDefaultInitialize:
+        // Defaulted: remove param label of it.
+        I = argumentLabelIDs.erase(I);
+        break;
+      case TupleShuffleExpr::Variadic: {
+        auto variadicArgsNum = args->getVariadicArgs().size();
+        if (variadicArgsNum == 0) {
+          // No arguments: Remove param label of it.
+          I = argumentLabelIDs.erase(I);
+        } else if (variadicArgsNum == 1) {
+          // One argument: Just advance.
+          ++I;
+        } else {
+          // Two or more arguments: Insert empty labels after the first one.
+          I = argumentLabelIDs.insert(++I, --variadicArgsNum, Identifier());
+          I += variadicArgsNum;
+        }
+        break;
+      }
+      default:
```

Comment:
We cannot use explicit `case` value here,
because the subject value is [basically a index of the arguments](https://github.com/rintaro/swift/blob/8f5fca013474dd8e00485a40f161e5761c2d8371/include/swift/AST/Expr.h#L2746-L2770).
I think the assertion below (`suffleIdx == (I - argumentLabelIDs.begin()` )
will catch any new kind of special value.
Am I missing something?